lib/model-interfaces/bedrock-agents/index.ts

```diff
@@ -0,0 +1,120 @@
+      code: props.shared.sharedCode.bundleWithLambdaAsset(
+        path.join(__dirname, "./functions/request-handler")
+      ),
+      handler: "index.handler",
+      runtime: props.shared.pythonRuntime,
+      architecture: props.shared.lambdaArchitecture,
+      tracing: lambda.Tracing.ACTIVE,
+      timeout: cdk.Duration.minutes(15),
+      memorySize: 1024,
+      logRetention: logs.RetentionDays.ONE_WEEK,
+      layers: [props.shared.powerToolsLayer, props.shared.commonLayer],
+      environment: {
+        ...props.shared.defaultEnvironmentVariables,
+        CONFIG_PARAMETER_NAME: props.shared.configParameter.parameterName,
+        SESSIONS_TABLE_NAME: props.sessionsTable.tableName,
+        SESSIONS_BY_USER_ID_INDEX_NAME: props.byUserIdIndex,
+        API_KEYS_SECRETS_ARN: props.shared.apiKeysSecret.secretArn,
+        MESSAGES_TOPIC_ARN: props.messagesTopic.topicArn,
+      },
+    });
+
+    props.sessionsTable.grantReadWriteData(requestHandler);
+    props.messagesTopic.grantPublish(requestHandler);
+    props.shared.apiKeysSecret.grantRead(requestHandler);
+    props.shared.configParameter.grantRead(requestHandler);
+
+    requestHandler.addToRolePolicy(
+      new iam.PolicyStatement({
+        actions: ["bedrock:InvokeAgent", "bedrock:GetAgentAlias"],
+        resources: ["*"],
+      })
+    );
+
+    if (props.config.bedrock?.roleArn) {
+      requestHandler.addToRolePolicy(
+        new iam.PolicyStatement({
+          actions: ["sts:AssumeRole"],
+          resources: [props.config.bedrock.roleArn],
+        })
+      );
+    }
+
+    const deadLetterQueue = new sqs.Queue(this, "DLQ", { enforceSSL: true });
+    const queue = new sqs.Queue(this, "Queue", {
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+      // https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#events-sqs-queueconfig
+      visibilityTimeout: cdk.Duration.minutes(15 * 6),
+      enforceSSL: true,
+      deadLetterQueue: {
+        queue: deadLetterQueue,
+        maxReceiveCount: 3,
+      },
+    });
+
+    queue.addToResourcePolicy(
+      new iam.PolicyStatement({
+        actions: ["sqs:SendMessage"],
+        resources: [queue.queueArn],
+        principals: [
+          new iam.ServicePrincipal("events.amazonaws.com"),
+          new iam.ServicePrincipal("sqs.amazonaws.com"),
+        ],
+      })
+    );
+
+    props.messagesTopic.addSubscription(
+      new SqsSubscription(queue, {
+        filterPolicyWithMessageBody: {
+          direction: sns.FilterOrPolicy.filter(
+            sns.SubscriptionFilter.stringFilter({
+              allowlist: [Direction.In],
+            })
+          ),
+          modelInterface: sns.FilterOrPolicy.filter(
+            sns.SubscriptionFilter.stringFilter({
+              allowlist: [ModelInterface.BedrockAgent],
+            })
+          ),
+        },
+      })
+    );
+
+    requestHandler.addEventSource(new lambdaEventSources.SqsEventSource(queue));
+
+    this.ingestionQueue = queue;
```

Comment:
I would recommend to add it to the dashboard
https://github.com/aws-samples/aws-genai-llm-chatbot/blob/3075d2c8f569a53fac16582b2655ff7976d17a4a/lib/aws-genai-llm-chatbot-stack.ts#L244

In a future step I will add alarms using these.